Fix tests and case where changes wouldn't be reported on missing-linted-files

nivekh opened the pull request:
### SUMMARY:

- Fixes some tests which weren't actually asserting
- Fixes mocks to include code files for 'is file still present' checks
- Fixes case where changes would not be reported if a file was missing from the lint but still present on disk 
   - This may not actually be intended behaviour if the test was written incorrectly?

### TESTING NOTES:

- Previously, some tests would run their expect-assert code within a for-loop, which doesn't actually work.
   - Instead, use `expect.to.include.members([1,2])`
- Previously, a test just ran as `expect(messages == [])`, which never actually asserts anything.
   - Instead, use `expect(messages.length).to.equal(0)`
   - Actually, because it prints the table, we need to expect 'no messages' to have one message - the inner formatter message

### SCREENSHOTS OR IT DIDN'T HAPPEN:

- It's tests. :)

Jmsa commented:
Hey @nivekh 👋 

Thanks for your work on this PR! I'll always welcome help cleaning up test assertions and reliability. FWIW, `expect`s in loops seem to work just fine here (let me know if you're seeing something different), but using `include.members` does result in much nicer error messages, so 👍 there.

One thing I could use more context on though is the change to report files that are present on disk but missing from the lint run. That behavior is intentional - were you running into an issue where it caused problems?

nivekh commented:
Previously, the test "doesn't update thresholds for files which weren't linted" would fail, because it expects to see `⚠️  eslint-ratchet: Changes to eslint results detected!!!` in the output.

Before that change, the case of <files missing from lint but otherwise no changes> wouldn't print that message. (This is what was making me think "oh, the for-loop expects must not be working right", which might not actually have been the case?)

If that was expected behaviour, then the test should be fixed instead. Let me know which way I should go on this and I'll amend the commits. 

Thanks for the review!

nivekh commented:
I've updated to revert to the previous behaviour, and removed the check for the message which shouldn't show.

As a heads-up, I'm cooking a slightly bigger PR over here: https://github.com/nivekh/eslint-formatter-ratchet/tree/use-formatter

General idea is letting users specify the formatter they want to use for the 'inner' formatted report, which prints out before the ratchet report. Dynamically loading formatters through the ESLint API is async, though, so ended up cleaning this stuff up while I was figuring out that.

Jmsa commented:
> I've updated to revert to the previous behaviour, and removed the check for the message which shouldn't show.

Nice. I didn't get a chance to respond earlier today but I think this is the right approach here. There might be cases where considering files outside the linted set makes sense - but I’d prefer to explore that as an option rather than the default since it seems less common.

Jmsa commented:
I've also merged in a change to unblock actions from running due to the deprecated cache@v2 so I'm going to rebase these changes so we can get checks to run (and I assume pass since they are locally for me).
